irmaclient/session.go

```diff
@@ -108,11 +110,20 @@ var supportedVersions = map[int][]int{
 		4, // old protocol with legacy session requests
 		5, // introduces condiscon feature
 		6, // introduces nonrevocation proofs
+		7, // introduces session binding
 	},
 }
 var minVersion = &irma.ProtocolVersion{Major: 2, Minor: supportedVersions[2][0]}
 var maxVersion = &irma.ProtocolVersion{Major: 2, Minor: supportedVersions[2][len(supportedVersions[2])-1]}
 
+func SetMaxVersion(version *irma.ProtocolVersion) {
```

Comment:
This function is only used by unittests that test legacy behaviour, and its presence would not be useful to users of this package; it might even be harmful. Is it possible to remove this from the public API without breaking those tests?

Reply:
No, because the tests are in another package and it need therefore to be accessed as an public function. These are tests that concern both server and irmaclient (sessiontest), so it is also a bit weird to move them to the irmaclient internal tests. We can include the maximum version in the initialization of the client. Then it is not so explicit, but still it is possible to change the maximum version then from the public API.

Reply:
I looked a bit in how to do this differently. It would be possible to include the protocol versions in the `Client` struct, such that we can also access them via special test handlers or via reflection. Downside is only that in all tests currently no `Client` instance is provided. This means I have to refactor all existing `session_test`'s to accept a client as a variable.

Then you would get for all `session_test` instances something like:
```
func testSigningSession(t *testing.T, client *irmaclient.Client, handler *TestClientHandler) {
...
```
and then have one general test that initiates a new client for each test and passes it to the subtests. 
```
func TestSession(t *testing.T) {
  t.Run("testSigningSession", useTestClient(testSigningSession))
 ...
}

func useTestClient(testFunc func(t *testing.T)) {
  client, handler := parseStorage(t)
  testFunc(t, client, handler)
}
```

The disadvantage is then that when at some point we remove the `legacy_test`'s, this kind of weird test structure remains or we have to refactor it back.

An easier alternative is then renaming the methods such that it is clear that they are for testing, like `SetTestMaxVersion` or something.

Reply:
Using reflection here like in the other test seems good.

Modifying every unit test like that would work, but it indeed seems a little excessive. Could the following also work?
- Add a new global variable `var maxClientVersion = &irma.ProtocolVersion{Major: 2, Minor: 7}` to `sessiontest`
- Make `parseStorage()` set that version on the client using reflection
- `TestWithoutBindingSupport` starts by setting `maxVersion` to 2.6 and ends (using `defer`) by switching it back